src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java

```diff
@@ -111,20 +112,39 @@ public Response getCollectionById(
       @PathParam("collectionId") Integer collectionId) {
     try {
       DarCollection collection = darCollectionService.getByCollectionId(collectionId);
-      User user = userService.findUserByEmail(authUser.getEmail());
-
-      if (user.hasUserRole(UserRoles.ADMIN) || checkSoPermissionsForCollection(user, collection)
-          || checkDacPermissionsForCollection(user, collection)) {
-        return Response.ok().entity(collection).build();
-      }
-      validateUserIsCreator(user, collection);
+      validateRequestingUserForDarCollection(authUser, collection);
       return Response.ok().entity(collection).build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
 
+  @GET
+  @Path("{collectionId}/electionHistory")
+  @Produces("application/json")
+  @PermitAll
+  public Response getCollectionWithAllElectionsByCollectionId(
+      @Auth AuthUser authUser,
```

Comment:
If you change this to a `DuosUser`, you'll get a full `User` object that you don't have to re-query for in the validation method